scripts/publish-docker-image.nix

```diff
@@ -0,0 +1,142 @@
+# This is run via a nix flake package:
+#
+#   $ nix run .#publish-docker-image <git-ref>
+#
+# The script is automatically checked with shellcheck, and run with bash using
+# a sensible set of options.
+{
+  # These arguments are passed explicitly
+  docker-images
+, format ? "oci"
+, registry ? { host = "ghcr.io"; repo = "hasura/ndc-mongodb"; }
+, target-protocol ? "docker://"
+
+  # These arguments are automatically populated from nixpkgs via `callPackage`
+, buildah
+, coreutils
+, git
+, writeShellApplication
+}:
+writeShellApplication {
+  name = "publish-docker-image";
+  runtimeInputs = [ coreutils git buildah ];
+  text = ''
```

Comment:
This is a new kind of crime I had not previously considered.

Reply:
Hey, it's really convenient to be able to interpolate nix values into the script! Plus if you have the right editor setup the shell script syntax highlighting and indentation work correctly.